### PR TITLE
Add AuthRequestor interface

### PR DESCRIPTION
**Pull request by chrisccoulson**
This adds an AuthRequestor interface that is used by the Activate*
APIs to request a credential (passphrase or recovery key) via some
agent.

It also adds an implementation of AuthRequestor that uses
systemd-ask-password. The new API provides a simple template based
way to define the message that is displayed.

### CODE REVIEW

File: auth_requestor_systemd.go [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+	cmd := exec.Command(
+		"systemd-ask-password",
+		"--icon", "drive-harddisk",
+		"--id", filepath.Base(os.Args[0])+":"+sourceDevicePath,
+		msg)
+	out := new(bytes.Buffer)
+	cmd.Stdout = out
+	cmd.Stdin = os.Stdin
+	if err := cmd.Run(); err != nil {
+		return "", err
```
Comment (pedronis) [inline auth_requestor_systemd.go:56]: this error path is not tested

**Review by pedronis — COMMENTED** [review]
looks good overall, small question about testing one main error path

File: auth_requestor_systemd.go (hunk 1 above)
Comment (chrisccoulson) [reply auth_requestor_systemd.go:56]: Thanks, I've added a test for this now.

**Review by pedronis — APPROVED** [review]
+1

**Pull request merged**